tensorflow/python/data/experimental/kernel_tests/take_while_test.py

```diff
@@ -83,6 +83,22 @@ def not_equal(string):
     with self.assertRaises(errors.OutOfRangeError):
       self.assertEqual(b"test", self.evaluate(next_element()))
 
+  def testTakewhileDatasetShortCircuit(self):
+    def _predicate_func(data_elem):
+      return data_elem
+
+    boolean_array = [True, True, True, True, False]
+    dataset = dataset_ops.Dataset.from_tensor_slices(boolean_array).apply(
+        take_while_ops.take_while(_predicate_func))
+
+    next_element = self.getNext(dataset)
+    self.assertEqual(True, self.evaluate(next_element()))
```

Comment:
```
for value in boolean_array:
  if value:
    self.assertTrue(self.evaluate(next_element()))
  else:
    with self.assertRaises(errors.OutOfRangeError):
      self.evaluate(next_element())
    break;
```

and change `boolean_array` to:

```
values = [True]*size
values[index] = False 
```

where `size` and `0 <= index < size` are parameters of the test, using 3-5 combinations

Reply:
I can instead do something like:

```python
boolean_array = [True]*size
boolean_array[index] = False

for _ in range(index):
  self.assertTrue(self.evaluate(next_element())

with self.assertRaises(errors.OutOfRangeError):
  self.evaluate(next_element())
```